### ISSUE TITLE
some problem about megaface evaluation

### ISSUE THREAD

**Issue by ReverseSystem001** [cavalleria/cavaface.pytorch#28]
When I evaluate the Mobilefacenet model, the megaface evaluation works well. But When i evaluate the Ghostnet, it reports error like:
-------------------------------------------------------------------------------
Total Tensors: 4096752 	Used Memory: 15.70M
The allocated memory on cuda:0: 77.55M
Memory differs due to the matrix alignment or invisible gradient buffer tensors
-------------------------------------------------------------------------------
min gpu free mem: 8000000000.0 B
min gpu free mem: 8000000000.0 B
min gpu free mem: 102000000 B
min gpu free mem: 162000000 B
Finish loading model /home/vision_rd/face_Recognition/models/GhostNet_Arcface/model/Epoch_24_Time_2020-07-21-13-27_checkpoint.pth, infer with shape: (198, 3, 112, 112)
Loading model time cost: 43.907608 seconds.

Extract on megaface...
Noisy faces of scrub: 605
Noisy faces of gallery: 707
Begin to extract embedding of scrub faces...
Finish Load path of faces: 0/3530
begin thread

Segmentation fault: 11

Stack trace:
  [bt] (0) /usr/local/lib/python3.6/site-packages/mxnet/libmxnet.so(+0x41a8280) [0x7f6276fd8280]
  [bt] (1) /lib64/libc.so.6(+0x363b0) [0x7f635a6483b0]
  [bt] (2) /lib64/libc.so.6(cfree+0x1c) [0x7f635a697ecc]
  [bt] (3) /usr/lib64/python3.6/site-packages/cv2/cv2.cpython-36m-x86_64-linux-gnu.so(+0x4eda39) [0x7f62b277da39]
  [bt] (4) /usr/lib64/python3.6/site-packages/cv2/cv2.cpython-36m-x86_64-linux-gnu.so(+0x168cd5) [0x7f62b23f8cd5]
  [bt] (5) /lib64/libpython3.6m.so.1.0(_PyCFunction_FastCallDict+0x147) [0x7f635b3ea167]
  [bt] (6) /lib64/libpython3.6m.so.1.0(+0x1507df) [0x7f635b4557df]
  [bt] (7) /lib64/libpython3.6m.so.1.0(_PyEval_EvalFrameDefault+0x3a7) [0x7f635b44a0f7]
  [bt] (8) /lib64/libpython3.6m.so.1.0(+0x14f987) [0x7f635b454987]
Eval model: /home/vision_rd/yangwenbo/face_Recognition/models/GhostNet_Arcface/model/Epoch_24_Time_2020-07-21-13-27_checkpoint.pth,24, done!
I doubt why I only changed the network model(has been trained). it report errors.

**Comment by cavalleria** [comment]
I re-evalualate the ghostnet and have no such bugs.
please use this scripts

MODEL="/workspace/results/GhostNet_Arcface/model_ep24_2.pth,24"
python -u ./main.py \
    --eval_sets="megaface" \
    --model_type=pytorch_fp32 \
    --gpus "0,1,2,3" \
    --net_scale "light" \
    --model_path=${MODEL}
echo "Eval model: ${MODEL}, done!"

MODEL="/workspace/results/GhostNet_Arcface/model_ep24_2.pth,24"
python -u ./main.py \
    --eval_sets="ijbc" \
    --model_type=pytorch_fp32 \
    --gpus "0,1,2,3" \
    --net_scale "light" \
    --model_path=${MODEL}
echo "Eval model: ${MODEL}, done!"

**Comment by ReverseSystem001** [comment]
THX. I found the reason now. when I run the Ghostnet.pt you provide. I works well. the differences is: when I save the checkpoint I opened the notes:
342                 if cfg['RANK'] % ngpus_per_node == 0:
344                     torch.save(backbone.module.state_dict(), os.path.join(MODEL_ROOT_NOTJIT, "Backbone_{}_Epoch_{}_Time_{}_checkpoint.pth".format(BACKBONE_NAME, epoch + 1, get_time())))
345                     save_dict = {'EPOCH': epoch+1,
346                                 'HEAD': head.state_dict(),
347                                 'OPTIMIZER': optimizer.state_dict()}
348                     torch.save(save_dict, os.path.join(MODEL_ROOT_NOTJIT, "Head_{}_Epoch_{}_Time_{}_checkpoint.pth".format(HEAD_NAME, epoch + 1, get_time()))) 

356                     ori_backbone.load_state_dict(backbone.module.state_dict())
357                     ori_backbone.eval()
358                     x = torch.randn(1,3,112,112).cuda()
359                     traced_cell = torch.jit.trace(ori_backbone, (x))
360                     torch.jit.save(traced_cell, os.path.join(MODEL_ROOT, "Epoch_{}_Time_{}_checkpoint.pth".format(epoch + 1, get_time())))

It seems like that it can not use torch.save and torch.jit.save at the same time. it is so strange and I don not know why

**Comment by cavalleria** [comment]
The model torch.jit.save saved can be evaluated without defining model file.

**Comment by ReverseSystem001** [comment]
yes, it requires install libtorch. or it will report C++ errors when use torch.jit.load().  THX

**Comment by ReverseSystem001** [comment]
I revalueted the model I had trained. it works. maybe it is the GPU memory problems. thx
